source/agora/test/Base.d

```diff
@@ -683,11 +684,15 @@ private mixin template TestNodeMixin ()
     private immutable(Block)[] blocks;
 
     ///
-    public this (Config config, Registry* reg, immutable(Block)[] blocks)
+    public this (Config config, Registry* reg, immutable(Block)[] blocks,
+        immutable(PreImageInfo)[] preimages)
     {
         this.registry = reg;
         this.blocks = blocks;
         super(config);
+
+        foreach (const ref preimage; preimages)
+            assert(this.enroll_man.addPreimage(preimage));
```

Comment:
We already have the `getEnrollmentManager` injection routine. Maybe we can use that instead?

Reply:
Hmm no it doesn't work. The issue is the enrollment manager must be instantiated first, then the ledger is instantiated (and it takes the manager in the ctor), then the ledger populates the validator set in the enrollment manager when loading the blocks.

So preimages need to be added after the enrollment manager and the ledger are constructed, but before `onValidatorsChanged` has been called.